lib/segment/src/segment.rs

```diff
@@ -83,15 +83,22 @@ impl Segment {
         vectors: NamedVectors,
     ) -> OperationResult<PointOffsetType> {
         check_vectors_set(&vectors, &self.segment_config)?;
-        let mut new_internal_index = 0;
+        let mut new_internal_index = None;
         for (vector_name, vector) in vectors {
             let vector_name: &str = &vector_name;
             let vector = vector.into_owned();
             let vector_data = &self.vector_data[vector_name];
-            new_internal_index = {
+            let internal_id = {
                 let mut vector_storage = vector_data.vector_storage.borrow_mut();
                 vector_storage.update_vector(old_internal_id, vector)
             }?;
+            if new_internal_index.is_none() {
+                new_internal_index = Some(internal_id);
+            } else {
+                debug_assert_eq!(new_internal_index, Some(internal_id));
```

Comment:
I propose to add some error log. Because if it happens, we get new "unexists point"